### PR TITLE
ModelParser: Implement JavaScope code generator.

### PR DESCRIPTION
**Pull request by azabbasi**
Unit tests will follow in the next PR.

### CODE REVIEW

File: sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/CodeWriter.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+     */
+    public void increaseIndent() {
+        this.indentedFileWriter.increaseIndent();
+    }
+
+    /**
+     * Decreases indent level.
+     */
+    public void decreaseIndent() {
+        this.indentedFileWriter.decreaseIndent();
```
Comment (barustum) [inline sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/CodeWriter.java:88]: does this method need a 'throws IOException'?

Comment (azabbasi) [reply sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/CodeWriter.java:88]: Not this one, decreaseIndent only decrements an integer value.

File: sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaLine.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+package com.azure.digitaltwins.parser.implementation.codegen;
+
+import java.io.IOException;
+
+public class JavaLine extends JavaStatement {
+
+    private String text;
+
+    /**
+     * Initializes a new instance of the {@link JavaLine} class.
```
Comment (barustum) [inline sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaLine.java:13]: having the @link here to JavaLine, wouldn't that just cause a navigation to the same page? so no-op?

File: sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaSorted.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+package com.azure.digitaltwins.parser.implementation.codegen;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * Generator for a series of java code lines whose code-generation order is determined by a lexicoraphical sort rather than the order in which they are added.
```
Comment (barustum) [inline sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaSorted.java:12]: lexico**g**raphical

**Review by barustum — APPROVED** [review]

File: sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaLine.java (hunk 2 above)
Comment (azabbasi) [reply sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/implementation/codegen/JavaLine.java:13]: Yeah, I use the link so the documentation would change if I decide to refactor the class name.

**Pull request merged**